cmd/git-init/main.go

```diff
@@ -82,15 +82,16 @@ func main() {
 			logger.Fatalf("Failed to change directory with path %s; err %v", path, err)
 		}
 	} else {
-		run(logger, "git", "init")
+		runOrFail(logger, "git", "init")
 	}
 
-	run(logger, "git", "remote", "add", "origin", *url)
-	err = run(logger, "git", "fetch", "--depth=1", "--recurse-submodules=yes", "origin", *revision)
-	if err != nil {
+	runOrFail(logger, "git", "remote", "add", "origin", *url)
```

Comment:
The reason we don't use `runOrFail` here is that it uses `logger.Fatalf` which will log and call `Exit(1)` 
We needed to try the `git pull` case when this fails

Reply:
@pivotal-nader-ziada not sure to follow, the rest of the `git` commands are all referencing the `origin` remote. If we failed to add it, nothing else will work.

Reply:
Yes, you are right, my bad! I thought it was the fetch